### ISSUE TITLE
fix(diagnostics): `vertical_velocity` integrates the z-ghost slab and takes a raw float mask; settle the [Nz+1] shape (#210)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#290]
Part of #275. Resolves #210.

## Summary

`vertical_velocity` (`finitevolx/_src/operators/diagnostics.py:558-617`) returns `[Nz+1, Ny, Nx]` with `w[0] = 0` and `w[k+1] = −Σ_{k'≤k} div_h[k']·dz`. The cumulative sum runs over **all** `Nz` levels including `k = 0` (the bottom z-ghost) and `k = Nz−1` (the top z-ghost), so `w[1]` equals the bottom-ghost divergence and `w[Nz]` includes the top ghost. That is correct only when the z-ghosts are zero, which is false for BC-filled ghosts, and the docstring is silent about it. It is also the only `[Nz+1, …]` array in the package (`Mask3D.w` is `[Nz, Ny, Nx]`, the top face of `T[k]`), and it takes `mask` as a raw float array unlike every other operator.

## Reproduction

```python
import jax.numpy as jnp
from finitevolx import CartesianGrid3D, vertical_velocity
Nz, Ny, Nx = 5, 6, 8
g = CartesianGrid3D.from_interior(Nx-2, Ny-2, Nz-2, Lx=1.0, Ly=1.0, Lz=1.0)
x = jnp.arange(Nx) * g.dx
u = jnp.broadcast_to(x, (Nz, Ny, Nx))      # du/dx = 1 at every level, including the ghost levels
v = jnp.zeros_like(u)
w = vertical_velocity(u, v, g.dx, g.dy, g.dz)
print(w[1, 2, 2], w[2, 2, 2])                # -1.0, -2.0 : w[1] is the divergence of the bottom ghost slab
```

The grid side is already convention "A" (`[Nz, Ny, Nx]` everywhere); only the docstrings in `grid/base.py`, `grid/cartesian.py`, `grid/spherical.py` still tell users W lives in a separate `[Nz+1, Ny, Nx]` array.

## Proposed fix (3 lines + docs)

- Integrate over interior levels only: `div_h = div_h.at[0].set(0)` (so `w(floor) = 0` holds regardless of ghost contents), `w = −cumsum(div_h[1:-1]) · dz`, return `[Nz, Ny, Nx]` aligned with `Mask3D.w` (drop the leading zero slab, or keep it and document that `w[k]` is the top face of `T[k]`).
- Accept `Mask3D` (use `mask.w`) like every other operator; keep the float-array path with a deprecation.
- Fix the three grid docstrings.
- Update `tests/test_diagnostics.py::TestVerticalVelocity` (they currently use zero z-ghosts only) and add a non-zero-ghost case.

## Acceptance criteria

- [ ] Repro gives `w[k] == 0` on the floor and the interior-only integral above it.
- [ ] Output shape `[Nz, Ny, Nx]`, matching `Mask3D.w`; CHANGELOG entry.
- [ ] `vertical_velocity(mask=Mask3D)` works; raw float path warns.
